niworkflows/interfaces/bids.py

```diff
@@ -255,6 +262,32 @@ def _run_interface(self, runtime):
             )
             self._results['out_file'].append(out_file)
             self._results['compression'].append(_copy_any(fname, out_file))
+
+            is_nii = out_file.endswith('.nii') or out_file.endswith('.nii.gz')
+            if self.inputs.check_hdr and is_nii:
+                nii = nb.load(out_file)
+                hdr = nii.header.copy()
+                curr_units = tuple([None if u == 'unknown' else u
+                                    for u in hdr.get_xyzt_units()])
+                curr_codes = (int(hdr['qform_code']), int(hdr['sform_code']))
+
+                # Default to mm, use sec if data type is bold
+                units = (curr_units[0] or 'mm', 'sec' if dtype == '_bold' else None)
+                xcodes = (1, 1)  # Derivative in its original scanner space
+                if self.inputs.space:
+                    xcodes = (4, 4) if self.inputs.space in STANDARD_SPACES \
+                        else (2, 2)
+
+                if curr_codes != xcodes or curr_units != units:
+                    self._results['fixed_hdr'][i] = True
+                    hdr.set_qform(hdr.get_qform(), xcodes[0])
+                    hdr.set_sform(hdr.get_sform(), xcodes[1])
```

Comment:
Can `qform` and `sform` be different? I think we should just set both to `nii.affine`.